blog/2022-11-01-cfd-protocol-explained.md

```diff
@@ -166,17 +178,17 @@ In full, the `commitment_transaction` can be spent with:
     1. A series of CETs unlocked by one of `olivia`'s `attestation`s.
     2. A `refund_transaction` that can only be published at least some time after `olivia` has published the relevant `attestation`.
     3. One `punishment_transaction` per party, which can be constructed unilaterally if one's counterparty publishes a revoked `commitment_transaction`.
-    This transaction allows the aggrieved party.
+    This transaction allows the aggrieved party to claim all the coins in the `dlc_output`.
 
 During `contract_setup` and every subsequent `rollover`, parties work together to add another spending condition to the `dlc_output`.
 They each generate a revocation keypair `(r, R)` and a publication keypair `(p, P)` and share the public keys with each other.
 The `dlc_output` is no longer a simple 2-of-2 multisig on their public keys `A` and `B`.
 It also includes:
 
-    1. A way for the maker to spend it with knowledge of `a`, `r_maker` and `p_maker`.
-    1. A way for the taker to spend it with knowledge of `b`, `r_taker` and `p_taker`.
+    1. A way for the maker to spend it with knowledge of their own secret key `a`, `r_taker` and `p_taker`.
+    1. A way for the taker to spend it with knowledge of their own secret key `b`, `r_maker` and `p_maker`.
 
-Instead of exchanging regular signatures on the `commitment_transactions` with each other, each constructs an adaptor signature which when encrypted and published would leak the other party's publishing secret key.
+Instead of exchanging regular signatures on the `commitment_transactions` with each other, each constructs an adaptor signature which when decrypted and published would leak the other party's publishing secret key.
```

Comment:
```suggestion
Instead of exchanging regular signatures on the `commitment_transactions` with each other, both construct an adaptor signature which when decrypted and published leaks the other party's publishing secret key.
```

Perhaps slightly simpler?

Also, shouldn't broadcasting commit leak _your own_ publishing key, and not the one from the other party?

Assuming I am the maker, I already know my own secret key and learned the revocation key during rollover. All I am missing is the other taker's publish key. If the taker broadcasts an old state, I need to be able to learn that key. Hence, shouldn't the sentence above say:

> which when decrypted and published leaks your own publishing secret key to the other party.

Reply:
What I'm trying to say is that the adaptor signature each of them constructs would leak the other party's publishing secret key when published by their counterparty.

I like your simplification. Maybe I'll rewrite this bit so that it's more obvious.